src/backend/distributed/executor/multi_utility.c

```diff
@@ -549,11 +549,15 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 	 */
 	if (IsA(parsetree, DropdbStmt))
 	{
+		const bool missingOK = true;
 		DropdbStmt *dropDbStatement = (DropdbStmt *) parsetree;
 		char *dbname = dropDbStatement->dbname;
-		Oid databaseOid = get_database_oid(dbname, false);
+		Oid databaseOid = get_database_oid(dbname, missingOK);
```

Comment:
Maybe rely on `dropDbStatement ->missing_ok` to avoid any edge cases where the user drops an existing database which doesn't have a valid oid (if that's ever possible 😄 ):

```suggestion
		Oid databaseOid = get_database_oid(dbname, dropDbStatement ->missing_ok);
```

Reply:
I think `dbcommands.c` -> `dropdb()` should be the single point we throw errors for `DROP DATABASE`, otherwise debugging becomes difficult. Looking at `dbcommands.c`, it seems that all databases have a valid oid.